### DISCUSSION TITLE
lesson 7: raise ConnectionError( ConnectionError: Status 403

### DISCUSSION THREAD

**Discussion by ace-brown** [smartcontractkit/full-blockchain-solidity-course-py#1421] [Q&A] [answered]
Hi.
I was doing the last deployment on the rinkeby on lesson 7 (lottery) at about 8:17:38 of the video. That the following error occurred.

`Running '\Tutorials\Web development\My projects\Blockchain projects\smartcontract_lottery\scripts\deploy_lottery.py::main'...
Transaction sent: 0x5dbdcc25b674bddab8742b800f92ac3ba508dd3cb374e753081c26891c6c6bb8
  Gas price: 1.159990906 gwei   Gas limit: 997092   Nonce: 48
  Lottery.constructor confirmed   Block: 10526261   Gas used: 906448 (90.91%)
  Lottery deployed at: 0x74ce3DBE7125D80A1b9b41E680f4Bb695442074A

Waiting for https://api-rinkeby.etherscan.io/api to process contract...
  File "C:\Users\andishe\.local\pipx\venvs\eth-brownie\lib\site-packages\brownie\_cli\run.py", line 51, in main
    return_value, frame = run(
  File "C:\Users\andishe\.local\pipx\venvs\eth-brownie\lib\site-packages\brownie\project\scripts.py", line 103, in run
    return_value = f_locals[method_name](*args, **kwargs)
  File "\Tutorials\Web, line projects\Blockchain, in line
    deploy_lottery()
  File "\Tutorials\Web, line projects\Blockchain, in line
    lottery = Lottery.deploy(
  File "C:\Users\andishe\.local\pipx\venvs\eth-brownie\lib\site-packages\brownie\network\contract.py", line 532, in __call__
    return tx["from"].deploy(
  File "C:\Users\andishe\.local\pipx\venvs\eth-brownie\lib\site-packages\brownie\network\account.py", line 557, in deploy
    contract.publish_source(deployed_contract, silent=silent)
  File "C:\Users\andishe\.local\pipx\venvs\eth-brownie\lib\site-packages\brownie\network\contract.py", line 388, in publish_source     
    raise ConnectionError(
ConnectionError: Status 403 when querying https://api-rinkeby.etherscan.io/api: <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1" />
<title>403 - Forbidden: Access is denied.</title>
<style type="text/css">

</style>
<script async src='/cdn-cgi/bm/cv/669835187/api.js'></script></head>
<body>
<div id="header"><h1>Server Error</h1></div>
<div id="content">
<div class="content-container"><fieldset>
<h2>403 - Forbidden: Access is denied.</h2>
<h3>You do not have permission to view this directory or page using the credentials that you supplied.</h3>
</fieldset></div>
</div>
<script type="text/javascript">(function(){window['__CF$cv$params']={r:'6fde2f8268806946',m:'5cca27c03939723f07ce9fb857df210db1d92859-1650293140-0-AX0yMNRHWa6Ihdm/qmvTgS5zmxuo6xkzvzl3BNoOL12fM//wY0Ny4Xlv3ZRP8fnWLzbkuVNmNbN2kMmlhFLAQuUmwv6hSA/XnQxpZVV6rq2l4lCkspvMtd923Cr/WeggI5wapb5yx4COk99pf+Y/5ik46E6mYIo72hcGZbE8wSfE6SH/S6qK6n0QRZrG3ke3p90C/IZdxHJSXQDRDFb6UuEMTl/krrkVpp7IcBGz7XPI0Hexeda967TzuKk1UhK8F1VpqeWGWpBmwUQkwAYKJ2/VVDdGwW6Vu93P8W3KlrIDL1JSbDX9RoztjRYU7VacWw==',s:[0x202d1adcd7,0x42148f67e9],}})();</script></body>
</html>
`

**Comment by cromewar** [comment, marked as answer]
Hello this means you don't have well configured the Etherscan API con your project, remember to add the .env file woth the API key and also if you are on windows add that key to PATH using windows advanced configuration.

**Reply by ace-brown** [reply]
Thanks. But after that it got another one, i guess it's related to link token. However i have link token in my metamask.

`Running '\Tutorials\Web development\My projects\Blockchain projects\smartcontract_lottery\scripts\deploy_lottery.py::main'...                                                                                                           :main'...
Transaction sent: 0x3b9e49bc0d8e5b6e7e4adf66270fab744cce367c1c0b7404b74a6e61b06fcb29
  Gas price: 1.075987411 gwei   Gas limit: 997092   Nonce: 60
  Lottery.constructor confirmed   Block: 10526485   Gas used: 906448 (90.91%)
  Lottery deployed at: 0x109f00c5976f5950A8923694b8af196D02c7a0EB

Waiting for https://api-rinkeby.etherscan.io/api to process contract...
Verification submitted successfully. Waiting for result...
Verification pending...
Verification pending...
Verification complete. Result: Pass - Verified
Deployed lottery!
Transaction sent: 0x04b60bc13c2dac5079cb5d03cd07af53db23a79f57331235e993039743c77d54
  Gas price: 1.075973723 gwei   Gas limit: 31572   Nonce: 61
  Lottery.startLottery confirmed   Block: 10526489   Gas used: 28702 (90.91%)

  Lottery.startLottery confirmed   Block: 10526489   Gas used: 28702 (90.91%)

The start of Lottery!
Transaction sent: 0x5f9b6ad76137718fae84a392072ae0aa9e4ba3beb702e773223b91334023d2b5
  Gas price: 1.075975198 gwei   Gas limit: 97460   Nonce: 62
  Lottery.enter confirmed   Block: 10526490   Gas used: 88600 (90.91%)

  Lottery.enter confirmed   Block: 10526490   Gas used: 88600 (90.91%)

Now you have entered the lottery!
Transaction sent: 0xd6a83bf179b8c6aa3796e1e3a24fcc1f50ff20656acf8816c02effe4f27f8041
  Gas price: 1.075976014 gwei   Gas limit: 56978   Nonce: 63
  LinkToken.transfer confirmed   Block: 10526491   Gas used: 51799 (90.91%)

  LinkToken.transfer confirmed   Block: 10526491   Gas used: 51799 (90.91%)

Funded the contract with link
  LinkToken.transfer confirmed   Block: 10526491   Gas used: 51799 (90.91%)

  File "C:\Users\andishe\.local\pipx\venvs\eth-brownie\lib\site-packages\brownie\_cli\run.py", line 51, in main
    return_value, frame = run(
  File "C:\Users\andishe\.local\pipx\venvs\eth-brownie\lib\site-packages\brownie\project\scripts.py", line 103, in run
    return_value = f_locals[method_name](*args, **kwargs)
  File "\Tutorials\Web, line projects\Blockchain, in line
    end_lottery()
  File "\Tutorials\Web, line projects\Blockchain, in line
    tx2 = lottery.endLottery({"from": account})
  File "C:\Users\andishe\.local\pipx\venvs\eth-brownie\lib\site-packages\brownie\network\contract.py", line 1710, in __call__
    return self.transact(*args)
  File "C:\Users\andishe\.local\pipx\venvs\eth-brownie\lib\site-packages\brownie\network\contract.py", line 1583, in transact
    return tx["from"].transfer(
  File "C:\Users\andishe\.local\pipx\venvs\eth-brownie\lib\site-packages\brownie\network\account.py", line 644, in transfer
    receipt, exc = self._make_transaction(
  File "C:\Users\andishe\.local\pipx\venvs\eth-brownie\lib\site-packages\brownie\network\account.py", line 727, in _make_transaction   
    raise VirtualMachineError(e) from None
  File "C:\Users\andishe\.local\pipx\venvs\eth-brownie\lib\site-packages\brownie\exceptions.py", line 93, in __init__
    raise ValueError(str(exc)) from None
ValueError: Gas estimation failed: 'execution reverted'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.`

**Comment by ace-brown** [comment]
You see, i had 60 link but now some has been deducted.
![image](https://user-images.githubusercontent.com/79329055/163834748-80c6c4c0-61ae-4be7-b707-e8be8b4c9730.png)
